### PR TITLE
More Fuzzy Symbol Fixes

### PR DESCRIPTION
**Pull request by kglowinski**
Fixes the corner case shown in the new test for fuzzy symbol mapping.

### CODE REVIEW

File: zipline/assets/assets.py [hunk 1]
```diff
@@ -657,8 +657,11 @@ def _lookup_symbol_fuzzy(self, symbol, as_of_date):
             return self.retrieve_asset(sid_keys[0])
 
         for sid, sym in options.items():
-            if sym == symbol:
-                # look for an exact match on the asof date
+            # Possible to have a scenario where multiple fuzzy matches have the
+            # same date. Want to find the one where symbol and share class
+            # match.
+            op_cs, op_sc = split_delimited_symbol(sym)
```
Comment (llllllllll) [inline zipline/assets/assets.py]: what do `op_cs` and `op_sc` stand for?

File: tests/test_assets.py [hunk 2]
```diff
@@ -757,6 +757,56 @@ def test_lookup_mult_are_one(self):
...
+            ]
+        )
+        self.write_assets(equities=df)
+        finder = self.asset_finder
+
+        # If we are able to resolve this with any result, means that we did not
+        # raise a MultipleSymbolError.
+        result = finder.lookup_symbol(
+            'FOO/B',
+            date + timedelta(90),
```
Comment (llllllllll) [inline tests/test_assets.py]: what units are these?

File: zipline/assets/assets.py (hunk 1 above)
Comment (kglowinski) [inline zipline/assets/assets.py]: option_company_symbol and option_share_class, but point taken. I'll fix that.

File: zipline/assets/assets.py [hunk 3]
```diff
@@ -657,8 +657,12 @@ def _lookup_symbol_fuzzy(self, symbol, as_of_date):
...
 
         for sid, sym in options.items():
-            if sym == symbol:
-                # look for an exact match on the asof date
+            # Possible to have a scenario where multiple fuzzy matches have the
+            # same date. Want to find the one where symbol and share class
+            # match.
+            option_company, option_share_class = split_delimited_symbol(sym)
+            if (company_symbol == option_company and
+                    share_class_symbol == option_share_class):
```
Comment (richafrank) [inline zipline/assets/assets.py]: Up to you, but could also do:
```python
if (company_symbol, share_class_symbol) == split_delimited_symbol(sym):
```
```

**Comment by llllllllll** [comment]
Rich had a style comment but lgtm
